NodeMediaServer

When trying to start the app with NodeMediaServer enabled, I get the following

> `> node ./lib/app
> 
> 
>     ███╗   ██╗ ██████╗  █████╗ ██╗     ██████╗ ███████╗
>     ████╗  ██║██╔═══██╗██╔══██╗██║     ██╔══██╗██╔════╝
>     ██╔██╗ ██║██║   ██║███████║██║     ██████╔╝███████╗
>     ██║╚██╗██║██║   ██║██╔══██║██║     ██╔══██╗╚════██║
>     ██║ ╚████║╚██████╔╝██║  ██║███████╗██████╔╝███████║
>     ╚═╝  ╚═══╝ ╚═════╝ ╚═╝  ╚═╝╚══════╝╚═════╝ ╚══════╝ v1.7.0
> 
> [6/4/2020] [3:32:18 PM] [OBS] › ℹ  info      Connecting & authenticating
> [6/4/2020] [3:32:18 PM] [CHT] › ℹ  info      Connecting to twitch
> /root/nginx-obs-automatic-low-bitrate-switching/lib/components/NodeMediaServer.js:20
>   console = _signale.default.scope("NMS");
>           ^
> 
> TypeError: Cannot set property console of #<Object> which has only a getter
>     at new NodeMediaServerManager (/root/nginx-obs-automatic-low-bitrate-switching/lib/components/NodeMediaServer.js:20:11)
>     at Object.<anonymous> (/root/nginx-obs-automatic-low-bitrate-switching/lib/app.js:22:23)
>     at Module._compile (module.js:652:30)
>     at Object.Module._extensions..js (module.js:663:10)
>     at Module.load (module.js:565:32)
>     at tryModuleLoad (module.js:505:12)
>     at Function.Module._load (module.js:497:3)
>     at Function.Module.runMain (module.js:693:10)
>     at startup (bootstrap_node.js:188:16)
>     at bootstrap_node.js:609:3
> 
> npm ERR! Linux 4.15.0-20-generic
> npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "start"
> npm ERR! node v8.10.0
> npm ERR! npm  v3.5.2
> npm ERR! code ELIFECYCLE
> npm ERR! nginx-obs-automatic-low-bitrate-switching@1.7.0 start: `node ./lib/app`
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the nginx-obs-automatic-low-bitrate-switching@1.7.0 start script 'node ./lib/app'.
> npm ERR! Make sure you have the latest version of node.js and npm installed.
> npm ERR! If you do, this is most likely a problem with the nginx-obs-automatic-low-bitrate-switching package,
> npm ERR! not with npm itself.
> npm ERR! Tell the author that this fails on your system:
> npm ERR!     node ./lib/app
> npm ERR! You can get information on how to open an issue for this project with:
> npm ERR!     npm bugs nginx-obs-automatic-low-bitrate-switching
> npm ERR! Or if that isn't available, you can get their info via:
> npm ERR!     npm owner ls nginx-obs-automatic-low-bitrate-switching
> npm ERR! There is likely additional logging output above.
> 
> npm ERR! Please include the following file with any support request:
> npm ERR!     /root/nginx-obs-automatic-low-bitrate-switching/npm-debug.log`

--- a/lib/components/NodeMediaServer.js
+++ b/lib/components/NodeMediaServer.js
@@ -22,7 +22,7 @@ var NodeMediaServerManager = function NodeMediaServerManager(nmsConfig, obsSwitc
   } catch (e) {
     console.log = _signale.default.scope("NMS").log;
 
-    _signale.default.scope("NMS").warn("Couldn't set console for node-media-server. Consider upgrading to node 10 or higher to resovle this problem (error: ".concat(e, ")"));
+    _signale.default.scope("NMS").warn("Couldn't set console for node-media-server. Consider upgrading to node 10 or higher to resolve this problem (error: ".concat(e, ")"));
   }
 
   this.obsSwitcher = obsSwitcher;

--- a/lib/components/NodeMediaServer.js
+++ b/lib/components/NodeMediaServer.js
@@ -17,7 +17,14 @@ var NodeMediaServerManager = function NodeMediaServerManager(nmsConfig, obsSwitc
     return;
   }
 
-  console = _signale.default.scope("NMS");
+  try {
+    console = _signale.default.scope("NMS");
+  } catch (e) {
+    console.log = _signale.default.scope("NMS").log;
+
+    _signale.default.scope("NMS").warn("Couldn't set console for node-media-server. Consider upgrading to node 10 or higher to resovle this problem (error: ".concat(e, ")"));
+  }
+
   this.obsSwitcher = obsSwitcher;
   this.nodeMediaServer = new _nodeMediaServer.default(nmsConfig);
   this.nodeMediaServer.run();

--- a/src/components/NodeMediaServer.js
+++ b/src/components/NodeMediaServer.js
@@ -8,7 +8,12 @@ class NodeMediaServerManager {
             return;
         }
 
-        console = signale.scope("NMS");
+        try {
+            console = signale.scope("NMS");
+        } catch(e) {
+            console.log = signale.scope("NMS").log;
+            signale.scope("NMS").warn(`Couldn't set console for node-media-server. Consider upgrading to node 10 or higher to resovle this problem (error: ${e})`);
+        }
 
         this.obsSwitcher = obsSwitcher;
 

--- a/src/components/NodeMediaServer.js
+++ b/src/components/NodeMediaServer.js
@@ -12,7 +12,7 @@ class NodeMediaServerManager {
             console = signale.scope("NMS");
         } catch(e) {
             console.log = signale.scope("NMS").log;
-            signale.scope("NMS").warn(`Couldn't set console for node-media-server. Consider upgrading to node 10 or higher to resovle this problem (error: ${e})`);
+            signale.scope("NMS").warn(`Couldn't set console for node-media-server. Consider upgrading to node 10 or higher to resolve this problem (error: ${e})`);
         }
 
         this.obsSwitcher = obsSwitcher;

